libs/execution_context/ebpf_maps.c

```diff
@@ -988,5 +1146,6 @@ ebpf_map_next_key(
     if (key_size != map->ebpf_map_definition.key_size) {
         return EBPF_INVALID_ARGUMENT;
     }
-    return ebpf_map_function_tables[map->ebpf_map_definition.type].next_key(map, previous_key, next_key);
+    // The code always inserts keys into all per-cpu maps, so it's sufficient to scan the first one.
```

Comment:
If I udnerstand right, this is optimizing for slow path performance instead of optimizing for hot path performance.   From UM you can do next_key for each CPU (from per CPU DPCs) and then take the lowest key among them, for example.

Reply:
Given that the bpf_update_elem and bpf_delete_elem helpers, the keys are required to be in synch, so we can simply read the CPU 0 one.

Reply:
Comment is now out of date.